src/compiler/checker.ts

```diff
@@ -11688,45 +11697,64 @@ namespace ts {
             return !!(type.flags & TypeFlags.Conditional || (type.flags & TypeFlags.Intersection && some((type as IntersectionType).types, isOrHasGenericConditional)));
         }
 
-        function elaborateError(node: Expression | undefined, source: Type, target: Type, relation: Map<RelationComparisonResult>, headMessage: DiagnosticMessage | undefined): boolean {
+        function elaborateError(

```

Comment:
the bulk of the lines in this PR go to making elaborateError and children handle `containingMessageChain` and `errorOutputContainer` the way that `checkTypeRelatedTo` already does. These two parameters are the mechanism for *returning* errors rather than immediately *logging* them, which allows me to build a tree of errors. That actually happens near the bottom of this change in checker.